### DISCUSSION TITLE
Energy correction

### DISCUSSION THREAD

**Discussion by kinkate** [TonyM1958/HA-FoxESS-Modbus#17] [Q&A]
In your code I found an absolut value for correcting the load-energy-value reading from inverter. Where did you get this value from or how did you figured out it? 
My inverter has also a wrong register-value, but after your correction I still have a load-energy-value of 118.643.367,4 kWh....

**Comment by TonyM1958** [comment]
I worked out the value by comparing the register value before and after it got corrupted but the algorithm is much simpler than it at first appears. I was intending to update it but haven't got around to it yet.

Try setting the value as follows: the value is read from the inverter as unsigned 32-bits with a scale factor of 10. The problem is that the high word of the 32-bits get corrupted but the lower 16-bits is correct. So, take the incorrect value, multiply by 10 and convert to hex. Zero out the lower 16-bits, convert that back to decimal and divide by 10. Now set the correction so it subtracts this value and you should find that fixes the values reported.

**Reply by TonyM1958** [reply]
Sorry, previously suggested value was wrong. Try using the value 118_639_820.8 as your correction.

**Reply by kinkate** [reply]
Hello Tony!
It works for me! You are my hero on this Sunday! Thanks a lot!
